extensions/cornerstone/src/hooks/useViewportActionCorners.ts

```diff
@@ -0,0 +1,123 @@
+import { useEffect } from 'react';
+import React from 'react';
+import { AllInOneMenu } from '@ohif/ui-next';
+import { getWindowLevelActionMenu } from '../components/WindowLevelActionMenu/getWindowLevelActionMenu';
+import { getViewportDataOverlaySettingsMenu } from '../components/ViewportDataOverlaySettingMenu';
+import { getViewportOrientationMenu } from '../components/ViewportOrientationMenu';
+
+/**
+ * Viewport action corner location type
+ */
+export interface ViewportActionCornerLocation {
+  type: 'VIEWPORT';
+  id: string;
+  corner?: 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right';
+}
+
+/**
+ * Viewport action corner service interface
+ */
+export interface ViewportActionCornerService {
+  addComponent: (params: {
+    viewportId: string;
+    id: string;
+    component: React.ReactNode;
+    location: ViewportActionCornerLocation | string;
+    indexPriority?: number;
+  }) => void;
+  clear: (viewportId: string) => void;
+  getAlignAndSide: (location: ViewportActionCornerLocation | string) => {
+    align: 'start' | 'center' | 'end';
+    side: 'top' | 'right' | 'bottom' | 'left';
+  };
+}
+
+/**
+ * Customization service interface
+ */
+export interface CustomizationService {
+  getCustomization: (name: string) => any;
+}
+
+/**
+ * Props for the useViewportActionCorners hook
+ */
+interface UseViewportActionCornersProps {
+  viewportId: string;
+  elementRef: React.MutableRefObject<HTMLDivElement>;
+  displaySets: AppTypes.DisplaySet[];
+  viewportActionCornersService: ViewportActionCornerService;
+  customizationService: CustomizationService;
+  commandsManager: AppTypes.CommandsManager;
+}
+
+/**
+ * Hook to manage viewport action corners for a Cornerstone viewport
+ */
+export function useViewportActionCorners({
+  viewportId,
+  elementRef,
+  displaySets,
+  viewportActionCornersService,
+  customizationService,
```

Comment:
Can you pass the system value in here instead of passing in any specific interfaces?  That is more reliably in terms of future changes, and is consistent for how to pass things around in terms of naming.